lib/src/ruby/fact.cc

```diff
@@ -120,6 +120,21 @@ namespace facter { namespace ruby {
                 // Set the value to what was resolved
                 _value = value;
                 _weight = weight;
+
+                if (! ruby.is_nil(_value)) {
```

Comment:
Some questions/comments on this:
* Currently when all custom facts have weight 0, we use the builtin fact value (see [here](https://github.com/puppetlabs/facter/blob/master/lib/src/ruby/fact.cc#L87-L99). Is this desired behavior?
* If ^ is the desired behavior, then this code will break that consistency when all of our custom facts with weight > 0 fail to resolve and one custom fact with weight == 0 resolves. Would that be bad?

Reply:
According to the docs facts should always have a positive integer value (i.e. not 0): https://docs.puppet.com/facter/3.0/fact_overview.html

So as a user I wouldn't expect a facter with weight == 0 to resolve over the builtin fact.

Reply:
I'd say any fact with weight 0 should be considered "disabled". If that ends up surprising users in the future it's easy to fix :man_shrugging: